plugins/snapshots/overlay/overlayutils/check.go

```diff
@@ -296,3 +300,45 @@ func SupportsIDMappedMounts() (bool, error) {
 
 	return true, nil
 }
+
+// SupportQuota checks if overlay filesystem supports quota or not.
+//
+// This function returns quotaSetter or error if it fails to check the filesystem.
+func SupportQuota(root string) (QuotaSetter, error) {
+	var backingFs = "<unknown>"
+	fileInfo, err := os.Stat(root)
+	if err != nil {
+		return nil, err
+	}
+	if !fileInfo.IsDir() {
+		root = filepath.Dir(root)
+	}
+
+	fsMagic, err := fs.GetMagic(root + "/")
+	if err != nil {
+		log.L.WithError(err).Warnf("get path %v filesystem magic number failed", root)
+		return nil, err
+	}
+
+	//TODO: check kernel version, kernel should be >= v4.5
+	if fsName, ok := fs.FsNames[fsMagic]; ok {
+		backingFs = fsName
+	}
+	switch fsMagic {
+	case fs.MagicXfs:
+		quotaCtl, err := quota.NewControl(root)
+		if err != nil {
+			log.L.WithError(err).Warnf("could not initinal quota control")
+			return nil, fmt.Errorf("directory '%s' does not support set quota, make sure mount with 'pquota'", root)
+		}
+		return func(targets []string, size uint64) error {
```

Comment:
noting we switched to a string... but are still using quota size here as a uint. thus no way to set "0" on size vs "" to mean unset return nil...